### DISCUSSION TITLE
Changing scope in provider settings.

### DISCUSSION THREAD

**Discussion by fellz** [strapi/strapi#7533] [Legacy Topics]
Hello
Scope for google authentication provider is hardcoded in `bootstrap.js` function. In Google case it's just `scope: ['email']`. It seems that i cann't customize this function( if i copy it to project from node_modules and edit it - nothing happens).  How can i setup another scope ? Whould be great if this can be set in **provider options** in admin panel.

https://github.com/strapi/strapi/blob/1e67421668a8a8528d03e09a52057bae35534e03/packages/strapi-plugin-users-permissions/config/functions/bootstrap.js#L43-L49

**Comment by derrickmehaffy** [comment]
You will need to use the [extensions system](https://strapi.io/documentation/v3.x/getting-started/troubleshooting.html#how-do-i-customize-a-plugin). Much like [adding your own provider](https://strapi.io/documentation/v3.x/plugins/users-permissions.html#adding-a-new-provider-to-your-project), but in this case we are editing an existing one.

You will need to copy that bootstrap.js file to `/extensions/users-permissions/config/functions/bootstrap.js`

**Reply by fellz** [reply]
> 
> 
> You will need to use the [extensions system](https://strapi.io/documentation/v3.x/getting-started/troubleshooting.html#how-do-i-customize-a-plugin). Much like [adding your own provider](https://strapi.io/documentation/v3.x/plugins/users-permissions.html#adding-a-new-provider-to-your-project), but in this case we are editing an existing one.
> 
> You will need to copy that bootstrap.js file to `/extensions/users-permissions/config/functions/bootstrap.js`

i did so but Providers.js takes data from database not from bootstrap.js. And just copying and editing not working here.

**Reply by derrickmehaffy** [reply]
Hi @fellz you are correct that it does store this information in the database but Strapi does compare the value within the bootstrap to that in the database during boot, I just tested using extensions on this file and it properly updated the value within the database:

![image](https://user-images.githubusercontent.com/8593673/91054677-af99a480-e5d8-11ea-88cd-341bb9953d24.png)
![image](https://user-images.githubusercontent.com/8593673/91054875-be805700-e5d8-11ea-9978-bada9c54c43d.png)

**Reply by negati-ve** [reply]
tested this, 
adding the above code to /extensions/users-permissions/config/functions/bootstrap.js 
does not update the database even when strapi is restart
